store/rootmulti/store.go

```diff
@@ -1035,8 +1035,14 @@ func commitStores(version int64, storeMap map[types.StoreKey]types.CommitKVStore
 	storeInfos := make([]types.StoreInfo, 0, len(storeMap))
 
 	for key, store := range storeMap {
-		commitID := store.Commit()
-
+		last := store.LastCommitID()
+		var commitID types.CommitID
+		if last.Version >= version {
+			last.Version = version
+			commitID = last
```

Comment:
I thought that the goal of the fix was to ensure a store that was behind, would be bumped to the correct version. This seems like the opposite...?

Reply:
This check is to avoid bump up [version](https://github.com/cosmos/cosmos-sdk/blob/358117ce5bacdfefe6f7b73659bc60689ce8df47/store/iavl/store.go#L131) by [commitStores](https://github.com/cosmos/cosmos-sdk/blob/358117ce5bacdfefe6f7b73659bc60689ce8df47/store/rootmulti/store.go#L438) when restart multi times before [flushMetadata](https://github.com/cosmos/cosmos-sdk/blob/358117ce5bacdfefe6f7b73659bc60689ce8df47/store/rootmulti/store.go#L439) finish.

Reply:
```
commitID = store.Commit()
if commitID.Version > version + 1 {
    commitID.Version = version + 1
}
```
Do you think sth like this would work? @alexanderbez

Reply:
> I thought that the goal of the fix was to ensure a store that was behind, would be bumped to the correct version. This seems like the opposite...?

it's to avoid a store's version bump higher than current version, when it's committed more times the normal.

Reply:
Ahhh I see now. Makes sense. Can we please add a comment in this block as to what and why we're doing this pls?